### PR TITLE
SUS-723 | comments index cleanup

### PR DESCRIPTION
**Pull request by rybmat**
Script for updating "deleted" flag in comments_index and also for removing entries from this table that has other namespace than supported by Wall/Forum associated in page table

### CODE REVIEW

File: extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php [hunk 1]
```diff
@@ -10,31 +10,66 @@ public function __construct() {
...
+	}
+
+	public function removeBrokenRows( DatabaseMysqli $dbMaster, DatabaseMysqli $dbSlave ) {
+		global $wgDBname;
+
+		$ids = $dbSlave->selectFieldValues(
 			[ 'page', 'comments_index' ],
 			'comment_id',
-			[ 'page_id=comment_id', 'page_namespace not in (500,501,1200,1201,2000,2001)' ],
+			[ 'page_id=comment_id', 'archived=0', 'deleted=0', 'removed=0', 'page_namespace not in (500,501,1200,1201,2000,2001)' ],
```
Comment (ghost) [inline extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php]: I would suggest using named constants for namespaces and `makeList()` to create a list for SQL `IN()` clause?

File: extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php [hunk 2]
```diff
@@ -10,31 +10,66 @@ public function __construct() {
...
+			}, $ids );
 
-			$this->output( "{$wgDBname}: " . count( $ids ) . " rows affected\n" );
+			$this->output( "{$wgDBname}: " . count( $ids ) . " rows to remove\n" );
 
 			if ( !$this->hasOption('dry-run') && !empty( $ids ) ) {
-				$db->delete(
+				$dbMaster->delete(
 					'comments_index',
 					[ "comment_id" => $ids ]
```
Comment (ghost) [inline extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php]: Please, add `__METHOD__` here. This deletes data which is logged in a more detailed manner. The source of the query is an important bit of information and helps to distinguish controlled deletion from unwanted incidents.

File: extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php (hunk 1 above)
Comment (mszabo-wikia) [inline extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php]: Is there a scenario where a page not in allowed namespace might be legitimately in `comments_index` table as archived/removed/deleted item?

**Review by ghost — CHANGES_REQUESTED** [review]

File: extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php [hunk 3]
```diff
@@ -10,31 +10,66 @@ public function __construct() {
...
+	}
+
+	public function updateDeletedFlag( DatabaseMysqli $dbMaster, DatabaseMysqli $dbSlave ) {
 		global $wgDBname;
 
-		$db = wfGetDB( DB_MASTER );
+		$ids = $dbSlave->selectFieldValues(
+			[ 'archive', 'comments_index' ],
+			'comment_id',
+			[ 'archived=0', 'deleted=0', 'removed=0', 'comment_id=ar_page_id', 'ar_namespace in (500,501,1200,1201,2000,2001)', 'comment_id not in (SELECT page_id from page)' ],
```
Comment (mszabo-wikia) [inline extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php]: I think it's enough to check `deleted` flag here. The other two aren't really delete options (thread closing/removal).

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

File: extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php (hunk 3 above)
Comment (rybmat) [reply extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php]: I didn't want to update by mistake rows that has one of those flags already set, so if it does not break anything I'd like to leave these checks as they are now

File: extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php (hunk 1 above)
Comment (rybmat) [reply extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php]: Yes it is. Just remove wall message with the delete flow that sets one of these flags and moves entry from page table to archive table (don't remember which flow it is) right after this create new article/upload file and Boom! New entry in page table have the same id as just removed wall message, but different namespace

Comment (rybmat) [reply extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php]: (auto increment on page_id works like max(page_id)+1 )

**Review by macbre — APPROVED** [review]

File: extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php [hunk 4]
```diff
@@ -18,7 +18,7 @@ public function execute() {
 		$updated = $this->updateDeletedFlag( $dbMaster, $dbSlave );
 		$deleted = $this->removeBrokenRows( $dbMaster, $dbSlave );
 
-		$this->output( "db:{$wgDBname} before:{$deleted['count']} updated:{$updated} deleted:{$deleted['deleted']}\n" );
+		$this->output( "db: {$wgDBname} before: {$deleted['count']} updated: {$updated} deleted: {$deleted['deleted']}\n" );
```
Comment (macbre) [inline extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php]: :+1:

**Pull request merged**